homeassistant/components/tami4/button.py

```diff
@@ -24,24 +26,46 @@ class Tami4EdgeButtonEntityDescription(ButtonEntityDescription):
     press_fn: Callable[[Tami4EdgeAPI], None]
 
 
-BUTTONS: tuple[Tami4EdgeButtonEntityDescription] = (
-    Tami4EdgeButtonEntityDescription(
-        key="boil_water",
-        translation_key="boil_water",
-        press_fn=lambda api: api.boil_water(),
-    ),
+@dataclass(frozen=True, kw_only=True)
+class Tami4EdgeDrinkButtonEntityDescription(ButtonEntityDescription):
+    """A class that describes Tami4Edge Drink button entities."""
+
+    press_fn: Callable[[Tami4EdgeAPI, Drink], None]
+
+
+BOIL_WATER_BUTTON = Tami4EdgeButtonEntityDescription(
+    key="boil_water",
+    translation_key="boil_water",
+    press_fn=lambda api: api.boil_water(),
 )
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Perform the setup for Tami4Edge."""
+
     api: Tami4EdgeAPI = hass.data[DOMAIN][entry.entry_id][API]
 
-    async_add_entities(
-        Tami4EdgeButton(api, entity_description) for entity_description in BUTTONS
-    )
+    async_add_entities([Tami4EdgeButton(api, BOIL_WATER_BUTTON)])
+
+    device = await hass.async_add_executor_job(api.get_device)
+    drinks = device.drinks
+
+    drink_buttons = [
+        Tami4EdgeDrinkButton(
+            api=api,
+            entity_description=Tami4EdgeDrinkButtonEntityDescription(
+                key=drink.id,
+                translation_key="prepare_drink",
+                translation_placeholders={"drink_name": drink.name},
+                press_fn=lambda api, drink: api.prepare_drink(drink),
+            ),
+            drink=drink,
+        )
+        for drink in drinks
+    ]
+    async_add_entities(drink_buttons)
```

Comment:
we still want to collect the entities in a list and call `async_add_entities` once